ci/Jenkins/nwxJenkins.groovy

```diff
@@ -1,34 +1,43 @@
 //BUILD_TAG is a unique build identifier provided by Jenkins
 
 def exportModules(buildModules){
-sh """
-set +x
-source /etc/profile
-module load ${buildModules}
-module save ${BUILD_TAG}
-"""
+    sh """
+       set +x
+       source /etc/profile
+       module load ${buildModules}
+       module save ${BUILD_TAG}
+       """
 }
 
-def compileRepo(repoName, doInstall, cmakeCommand){
+def getCPP(){
+    sh """
+    set +x
+    source /etc/profile
+    module load cmake
+    git clone https://github.com/CMakePackagingProject/CMakePackagingProject
+    cd CMakePackagingProject
+    cmake -H. -Bbuild -DBUILD_TESTS=OFF \
+                      -DCMAKE_INSTALL_PREFIX=${WORKSPACE}/install
+    cmake --build build --target install
+    """
+}
+
+def compileRepo(repoName){
     def installRoot="${WORKSPACE}/install"
     sh """
        set +x
-	source /etc/profile
-	module restore ${BUILD_TAG}
-        buildTests="True"
-        makeCommand=""
-        if [ ${doInstall} == "True" ];then
-            buildTests="False"
-            makeCommand="install"
-        fi
-	CC=gcc
-	CXX=g++
-        cmake -H. -Bbuild -DBUILD_TESTS=\${buildTests} \
-                          -DCMAKE_INSTALL_PREFIX=${installRoot}\
-                          -DCMAKE_PREFIX_PATH=${installRoot}\
-			  ${cmakeCommand}
-        cd build && make \${makeCommand}
-    """
+	   source /etc/profile
+	   module restore ${BUILD_TAG}
+	   if [ -d build ]; then
+	       rm -rf build
+	   fi
+       cmake -H. -Bbuild -DBUILD_TESTS=TRUE \
+                         -DCMAKE_INSTALL_PREFIX=${installRoot}\
+                         -DCMAKE_PREFIX_PATH=${installRoot} \
+                         -DCMAKE_CXX_COMPILER=g++ \
+                         -DCMAKE_C_COMPILER=gcc
```

Comment:
I had to default C/CXX compilers to GNU because of an issue with Hunter (https://github.com/NWChemEx-Project/DeveloperTools/pull/32)

That shouldn't be necessary anymore, so could you replace those cmake flags with ${cmakecommand}? That will let us build dependencies with the same compiler as the target repo, and gives a spot to specify additional cmake flags on a per-repo basis.

Reply:
To me `${cmakeCommand}` was misleading as I'd expect that to be the command to run CMake and not the flags to CMake.  I agree they shouldn't be hard coded so I made them arguments to the function.  To populate the values I opted to parse the module list, taking the appearance of a module for loading  the Intel compilers as a sign that the user wants to use `icc` and `icpc`.  We could add another check for Clang eventually.